### PR TITLE
Use middleman redirect to send package uris to preexisting latest version uris.

### LINKED ISSUE #118: Redirect https://hub.getdbt.com/<package> to https://hub.getdbt.com/<package>/latest
I went to https://hub.getdbt.com/fishtown-analytics/dbt_utils/ and got a "page not found", whereas I expected it to take me to https://hub.getdbt.com/fishtown-analytics/dbt_utils/latest

### PR DESCRIPTION
**Pull request by VersusFacit**
# Purpose

1️⃣  Closes #118
1 line solution that comes from a few iterations of experimentation. This address the the problem idiomatically and in a well-defined way.

I heartily invite you to inspect the behavior for yourself on the Netlify deploy :)

2️⃣  Documentation additions
Added documentation for how I got local development working on Big Sur.

# Impact

Redirect to latest version of a package, with no duplicating of uri "space" (cf. search space). This solution in particular requires no intervention on the web server layer yet keeps uris sensible for SEO tracking and such.

## Dev notes

Here's a trivial solution: "Just" add this code in `config.rb` (the equivalent in Middleman apps to the more ubiquitous, Rails-style `routes.rb`):
```ruby
      proxy "/#{package_name}/",
            '/package.template.html',
            :content_type => 'text/html',
            :layout => 'layout',
            :locals => {
              :package => package,
              :version => package.versions[package.latest]
            }
```

##### Caveat: package_name is actually `<org-name>/<package-name>`. Don't be fooled like I was at first!

One problem--when you reflect on it, this isn't a true redirect to the `latest` uri of a package. Instead, this exact uri `/#{package_name}/` is being invented with directions to call the same function as other already established uris. This is artificial. Here's some evidence (for the worse): to have `/#{package_name}/` AND `/#{package_name}` work, you must repeat this code a _second_ time for none /-suffix `/#{package_name}/` uris`. Welcome to the same SEO metric split across 3 dimensions.

To get that 🔥 301-like action, Middleman provides a could-be-better-documented [`redirect`](https://middlemanapp.com/basics/redirects/) builtin. This intercepts the uri and reroutes to `/#{package_name}/latest` (no duplication, or in this case triplication, of url search space needed). Hence, the redirect solution, with some clever trial and error to discover the exact syntax needed to redirect

TL;DR Middleman wants files to be redirected files, so to define behavior on directory uris, use the `index.html` behavior of HTTPS.

### CODE REVIEW

**Review by jasnonaz — APPROVED** [review]
Looks good and validated - appears to be behaving as expected.
